core/chains/evm/headtracker/head_saver.go

```diff
@@ -14,7 +15,7 @@ type headSaver struct {
 	orm    ORM
 	config Config
 	logger logger.Logger
-	heads  Heads
+	heads  htrktypes.Heads[*evmtypes.Head, common.Hash]
```

Comment:
Since HeadSaver is already part of the same package, it doesn't need to use the exported interface.
It can directly refer to the heads struct, and use unexported methods.

Reply:
I might be misunderstanding this, but what do you mean by not using the exported interface and using the `head` struct directly? 


do you mean something like
```
type headSaver struct {
	orm    ORM
	config Config
	logger logger.Logger
	heads  *heads
}
```